backbone.js

```diff
@@ -613,43 +629,48 @@
 
       options = _.extend({validate: true, parse: true}, options);
       var wait = options.wait;
+      var success = options.success || _.identity;
+      var error = options.error;
+      delete options.success;
+      delete options.error;
 
       // If we're not waiting and attributes exist, save acts as
       // `set(attr).save(null, opts)` with validation. Otherwise, check if
       // the model will be valid when the attributes, if any, are set.
       if (attrs && !wait) {
-        if (!this.set(attrs, options)) return false;
+        if (!this.set(attrs, options)) return Promise.reject(this);
```

Comment:
I've just seen @jridgewell's take on this: [backbone-promise](https://github.com/jridgewell/backbone/blob/6de1cbbf0c13bc26a07a389ac2c955e0bf7b58bc/backbone.js#L604). And I think I've been utterly confused on this... For some reason I had it in my head that `false` meant that there was no change to the attributes... :-$